### DISCUSSION TITLE
MSS310 Power/Current

### DISCUSSION THREAD

**Discussion by Matt2uk** [krahabb/meross_lan#152] [General]
Hi. I installed 2 days ago and this is a great integration. However I can't seem to get my head round the values. 
Power looks roughly right based on what I have plugged in, voltage seems a bit on the high side but probably near enough (250v, id expect 230-240v for here in the uk). The current value hovever is way out from what I'd expect.
My guess is that it's somehow being back calculated from the Power but using 110V instead so I'm not sure if this is an issue that's internal to the unit or whether it's part of the integration?

Anyone noticed similar or have any ideas?
Cheers, Matt.

**Comment by krahabb** [comment]
Hello @Matt2uk ,
meross_lan does not perfomr any internal calculation for the current value except adjusting its scale by a fixed (1000) factor.
I've noticed the device has the ability to report what looks like a 'configuration' payload reporting some voltage and current scaling factors but I've never been able to modify/update it.
If you can enable tracing on your device (on the integration configuration panel) and upload it I could try inspect its configuration and see if it's any different from mine and maybe get some clues. 
My mss310 trace reports this for instance:
```text
2021/09/09 - 18:54:19	mqtt	GET	Appliance.Control.ConsumptionConfig	{"consumptionconfig": {}}
2021/09/09 - 18:54:19	mqtt	GETACK	Appliance.Control.ConsumptionConfig	{"config": {"voltageRatio": 188, "electricityRatio": 100}}
```
